### ISSUE TITLE
Terminating a blocked worker cause `terminate` to block

### ISSUE THREAD

**Issue by irbull** [denoland/deno#13705]
This bug is a follow up to https://github.com/denoland/deno/issues/13591, but a separate issue in my opinion. It's not platform specific like #13591, and it can cause the entire runtime to hang.

If a worker is currently blocked (in this case, trying to acquire a lock) and the main thread calls `terminate()`, terminate will block and no other code will be executed on the main thread.

Below is a snippet that demonstrates this. It creates a worker that tries to acquire an exclusive lock on `blocking-worker.ts` (make sure you have a file with that name in your working directory). The worker then waits. A second worker is created 500ms later that does the same thing. This worker fails to acquire the lock (as expected), and blocks. 

Then, the `main` thread tries to call `terminate()` on this (second) blocked worker. At this point the main thread blocks and no more heartbeats are sent.  You need the `unstable` flag to use the file locking API, but I think that's secondary. I tested this on 🦖 Deno 1.19.0 on Mac. I think that `terminate()` should be guaranteed to return and not block. To run this use:

```
deno run --allow-read --unstable blocking-worker.ts  
``` 

```
// blocking-worker.ts
const WORKER_JS = `
    const file = await Deno.open('blocking-worker.ts');
    console.log('trying to lock: ', file.rid);
    Deno.flockSync(file.rid, true);
    console.log('locked');
    await new Promise(() => {}); // wait for termination
`;

// heartbeat every 400ms
setInterval(() => console.log('heartbeat'), 400);

const workerJS = URL.createObjectURL(
  new Blob([WORKER_JS], { type: 'application/javascript' }),
);

const startWorker = () =>
  new Worker(workerJS, {
    type: 'module',
    deno: {
      namespace: true,
    },
  });

// Start the first worker
startWorker();

// wait 500ms
await new Promise((resolve) => setTimeout(resolve, 500));

// start the second worker. This worker will fail to get a lock to a shared
// file and it will block.
const secondWorker = startWorker();

// Terminate the second worker while it's blocking
setTimeout(() => {
  console.log('Terminating the second worker');
  secondWorker.terminate();
  console.log('Terminated');
}, 1000);
```

**Comment by irbull** [comment]
Is there anything I can do to help with this issue? The blocking of the main thread when a blocked worker is `terminated` leads to a highly unstable system. Andreu had a suggestion on Discord about using a vec of JoinHandles per thread that are joined after the event loop finishes but before the isolate is dropped. https://discord.com/channels/684898665143206084/778060818046386176/944003162984030238

**Comment by andreubotella** [comment]
I run into a bit of an issue when working on it, and then I haven't had much time. I'll try to get the fix ready before 1.20
